### PR TITLE
AWS Lambda functionality

### PR DESCRIPTION
**Pull request by cbednarski**
- Includes documentation
- Includes acceptance tests

### CODE REVIEW

**Comment by catsby** [comment]
Reviewing now; looks like it needs a rebase though, do you have time? I can swoop in if you need

File: builtin/providers/aws/resource_aws_lambda_function.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+// CreateFunction in the API / SDK
+func resourceAwsLambdaFunctionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	functionName := d.Get("function_name").(string)
+	iamRole := d.Get("role").(string)
+
+	log.Printf("[DEBUG] Creating Lambda Function %s with role %s", functionName, iamRole)
+
+	zipfile, err := ioutil.ReadFile(d.Get("filename").(string))
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_lambda_function.go]: Does `ioutil.ReadFile` handle all the path expansion(s) we may need? e.g. does `~/some_path/file.zp` also work?

File: builtin/providers/aws/resource_aws_lambda_function.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+				time.Sleep(2 * time.Second)
+				continue
+			}
+		}
+		break
+	}
+	if err != nil {
+		return fmt.Errorf("Error creating Lambda function: %s", err)
+	}
+	// Add retry wrapper around this
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_lambda_function.go]: Is this comment still relevant? Looks like there's a retry above, maybe it was for that?

File: builtin/providers/aws/resource_aws_lambda_function_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+	}
+
+	return nil
+
+}
+
+func testAccCheckAwsLambdaFunctionExists(n string, function *lambda.GetFunctionOutput) resource.TestCheckFunc {
+	// Wait for IAM role
+	time.Sleep(100 * time.Millisecond)
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_lambda_function_test.go]: Why do we `sleep` here? I thought the _waiting for IAM role_ was handled in the resources `create` method?

**Comment by catsby** [comment]
Some nitpick/comments, other than than :+1:

File: builtin/providers/aws/resource_aws_lambda_function_test.go (hunk 3 above)
Comment (cbednarski) [inline builtin/providers/aws/resource_aws_lambda_function_test.go]: Good catch!

**Comment by cbednarski** [comment]
Merged / updated with master to get the awslabs -> aws change. Lemme know if you want me to rebase.

File: builtin/providers/aws/resource_aws_lambda_function.go (hunk 1 above)
Comment (cbednarski) [inline builtin/providers/aws/resource_aws_lambda_function.go]: Nope! Good catch on this.

**Comment by mitchellh** [comment]
OH YEAH!

**Comment by phinze** [comment]
:dancer:

**Pull request merged**